File: BookStore.App/wwwroot/js/Company.js

$(document).ready(function () {
    loadDataTable();
});

function loadDataTable() {
    $("#myTable").DataTable({
        ajax: {
            url: "/Admin/Companies/GetAll",
        },
        columns: [
            { data: "name" },
            { data: "phoneNumber" },
            { data: "address" },
            { data: "city" },
            { data: "state" },
            { data: "postalCode" },
            {
                data: "id",
                render: function (data) {
                    return `<a href="/admin/companies/Upsert/${data}" class="text-decoration-none">
                                <i class="bi bi-pencil-square"></i> Edit
                            </a> |
                            <a href="/admin/companies/delete/${data}" onclick="del()" class="text-decoration-none text-danger">
                                <i class="bi bi-trash3"></i> Delete
                            </a>`;
                },
            },
        ],
    });
}

function del() {
    var decision = confirm("are you sure?");
    if (decision == false) {
        event.preventDefault();
    }
};

File: BookStore.Models/Company.cs
namespace BookStore.Models
{
    public class Company
    {
        public int Id { get; set; }

        [Required]
        [StringLength(100)]
        public string? Name { get; set; }

        [Required]
        [StringLength(200)]
        public string? Address { get; set; }

        [Required]
        [StringLength(50)]
        public string? City { get; set; }

        [Required]
        [StringLength(50)]
        public string? State { get; set; }

        [Required]
        [StringLength(20)]
        [Display(Name = "Postal Code")]
        public string? PostalCode { get; set; }

        [Required]
        [StringLength(20)]
        [Display(Name = "Phone")]
        public string? PhoneNumber { get; set; }
    }
}
